Array

Hogjiv opened the pull request:
Look at my work

abrakadobr committed:
Merge pull request #11 from bluep-js/dev

0.3.7

Hogjiv committed:
Arrays added

abrakadobr commented on src/nodes/array/at.js:
```diff
@@ -0,0 +1,75 @@
+module.exports = ArrayAt
+
+
+/* 
```
no comments :)

abrakadobr replied:
comments :)

abrakadobr commented on src/nodes/array/concat.js:
```diff
@@ -0,0 +1,68 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments )

abrakadobr commented on src/nodes/array/copywithin.js:
```diff
@@ -0,0 +1,64 @@
+
+  static metadata() {
+    return {
+      name: 'CopyWithin',
```
names are "human readable" - so should be "Copy Within" with space between words

abrakadobr commented on src/nodes/array/copywithin.js:
```diff
@@ -0,0 +1,64 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
and comments

abrakadobr commented on src/nodes/array/each.js:
```diff
@@ -0,0 +1,73 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/entries.js:
```diff
@@ -0,0 +1,53 @@
+    this.debug('execute', inputs.array)
+    if (Array.isArray(inputs.array)) {
+      const res = inputs.array.entries()
+      for (let element of res) {
```
no console logs

abrakadobr commented on src/nodes/array/entries.js:
```diff
@@ -0,0 +1,53 @@
+const dayjs = require('dayjs')
```
this is iterator method (The entries() method returns a new Array Iterator object ) - we don't use iterators right now, so this file is not required

abrakadobr commented on src/nodes/array/filter.js:
```diff
@@ -0,0 +1,55 @@
+const dayjs = require('dayjs')
```
remove this one. we don't have function arguments right now

abrakadobr commented on src/nodes/array/includes.js:
```diff
@@ -0,0 +1,51 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/includes.js:
```diff
@@ -0,0 +1,51 @@
+  }
+
+  async execute(inputs) {
+    this.debug('execute', inputs) 
```
add check for input is array

abrakadobr commented on src/nodes/array/concat.js:
```diff
@@ -0,0 +1,68 @@
+  async execute(inputs) {    
+    let ret = inputs.array
+    if (Array.isArray(inputs.array)) { Object.keys(inputs).forEach(inp => {
+      if (inp.startsWith('array2')) ret = ret.concat(inputs[inp])
```
add check for isArray

abrakadobr commented on src/nodes/array/indexOf.js:
```diff
@@ -0,0 +1,57 @@
+
+  async execute(inputs) {
+    this.debug('execute', inputs.array )
+    let a = inputs.array.indexOf(inputs.element, inputs.index)
```
add check for isArray

abrakadobr commented on src/nodes/array/isArray.js:
```diff
@@ -0,0 +1,45 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/join.js:
```diff
@@ -0,0 +1,55 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/lastIndexOf.js:
```diff
@@ -0,0 +1,56 @@
+
+  static metadata() {
+    return {
+      name: 'LastIndexOf',
```
name with spaces

abrakadobr commented on src/nodes/array/lastIndexOf.js:
```diff
@@ -0,0 +1,56 @@
+  async execute(inputs) {
+    this.debug('execute', inputs.array )
+    if (Array.isArray(inputs.array)) {
+      const res = inputs.array.lastIndexOf(inputs.fromElement || inputs.fromIndex);
```
not correct. check docs

abrakadobr commented on src/nodes/array/length.js:
```diff
@@ -0,0 +1,46 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/length.js:
```diff
@@ -0,0 +1,46 @@
+  async execute(inputs) {
+    this.debug('execute', inputs)
+    const a = inputs.array
+    this.setOutput('result', a.length)
```
check for isArray

abrakadobr commented on src/nodes/array/pop.js:
```diff
@@ -0,0 +1,48 @@
+
+  async execute(inputs) {
+    this.debug('execute', inputs.array)
+    if (inputs.array) {
```
check for isArray

abrakadobr commented on src/nodes/array/reverse.js:
```diff
@@ -0,0 +1,47 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/reverse.js:
```diff
@@ -0,0 +1,47 @@
+
+  async execute(inputs) {
+    this.debug('execute', inputs) 
+    this.setOutput('result', inputs.array.reverse())
```
check for isArray

abrakadobr commented on src/nodes/array/shift.js:
```diff
@@ -0,0 +1,44 @@
+    return {
+      name: 'Shift',
+      code: 'array/shift',
+      type: 'modifier',
```
shift, pop, push, unshift are not modifiers.
they change values of inputs, so they should be callables and has execution flow input/output

abrakadobr commented on src/nodes/array/pop.js:
```diff
@@ -0,0 +1,48 @@
+    return {
+      name: 'Pop',
+      code: 'array/pop',
+      type: 'modifier',
```
shift, pop, push, unshift are not modifiers.
they change values of inputs, so they should be callables and has execution flow input/output

abrakadobr commented on src/nodes/array/slice.js:
```diff
@@ -0,0 +1,62 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/slice.js:
```diff
@@ -0,0 +1,62 @@
+}
+
+module.exports = ArraySlice
+/* 
```
comments

abrakadobr commented on src/nodes/array/slice.js:
```diff
@@ -0,0 +1,62 @@
+
+  async execute(inputs) {
+    this.debug('execute', inputs.valA, inputs.valB)
+    this.setOutput('result', inputs.array.slice(inputs.valA, inputs.valB))
```
add check for isArray

abrakadobr commented on src/nodes/array/slice.js:
```diff
@@ -0,0 +1,62 @@
+const dayjs = require('dayjs')
```
remove dayjs

abrakadobr commented on src/nodes/array/some.js:
```diff
@@ -0,0 +1,73 @@
+const dayjs = require('dayjs')
```
remove file, we don't have functions as variables yet

abrakadobr commented on src/nodes/array/splice.js:
```diff
@@ -0,0 +1,59 @@
+    return {
+      name: 'Splice',
+      code: 'array/splice',
+      type: 'modifier',
```
change values of inputs, so  should be callable and has execution flow input/output

abrakadobr commented on src/nodes/array/splice.js:
```diff
@@ -0,0 +1,59 @@
+      templates: {
+        A: {
+          allow: ['*'] // ,
+          // disallow: [],
```
comments

abrakadobr commented on src/nodes/array/splice.js:
```diff
@@ -0,0 +1,59 @@
+
+  async execute(inputs) {
+    this.debug('execute', inputs )
+    this.setOutput('result', inputs.array.splice(inputs.start, inputs.item))
```
check for isArray

abrakadobr commented on src/nodes/array/toString.js:
```diff
@@ -0,0 +1,47 @@
+const dayjs = require('dayjs')
```
remove dayjs

abrakadobr replied:
just remove all file :)

abrakadobr commented on src/nodes/array/toString.js:
```diff
@@ -0,0 +1,47 @@
+
+  static metadata() {
+    return {
+      name: 'Tostring',
```
human readable name

abrakadobr commented on src/nodes/array/unshift.js:
```diff
@@ -0,0 +1,61 @@
+    return {
+      name: 'Unshift',
+      code: 'array/unshift',
+      type: 'modifier',
```
shift, pop, push, unshift are not modifiers.
they change values of inputs, so they should be callables and has execution flow input/output

abrakadobr commented on src/nodes/array/unshift.js:
```diff
@@ -0,0 +1,61 @@
+          allow: ['*'] 
+        }
+      },
+      multiples: {
```
no multiples

abrakadobr commented on src/nodes/array/unshift.js:
```diff
@@ -0,0 +1,61 @@
+
+  async execute(inputs) {
+    this.debug('execute', inputs.array )
+    let a = inputs.array.unshift(inputs.element)
```
check for isArray

abrakadobr commented on src/nodes/array/values.js:
```diff
@@ -0,0 +1,50 @@
+const dayjs = require('dayjs')
```
remove file. we don't work with iterators

abrakadobr reviewed (commented):
keep going :)

Hogjiv committed:
Arrays fixed according comments.
Array.push was added
added few fixing in:
string.toLowerString, length,
float.gt,
boolean xor
